From 353e2d04075bb6597fcc4e327344d097d34a7df3 Mon Sep 17 00:00:00 2001
From: Zengruan Ye <yezengruan@huawei.com>
Date: Tue, 15 Mar 2022 15:46:07 +0800
Subject: [PATCH] arm/arm64: paravirt: Remove GPL from pv_ops export
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 353e2d04075bb6597fcc4e327344d097d34a7df3
Modified-by-SEL: No


virt inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I4VZPC
CVE: NA

--------------------------------

Commit 63042c58affc ("KVM: arm64: Add interface to support vCPU
preempted check") introduced paravirt spinlock operations, as pv_lock_ops
was exported via EXPORT_SYMBOL(), while the pv_ops structure containing
the pv lock operations is exported via EXPORT_SYMBOL_GPL().

Change that by using EXPORT_SYMBOL(pv_ops) for arm/arm64, as with the x86
architecture changes, the following:
https://lore.kernel.org/all/20181029150116.25372-1-jgross@suse.com/T/#u

Fixes: 63042c58affc ("KVM: arm64: Add interface to support vCPU preempted
check")
Signed-off-by: yezengruan <yezengruan@huawei.com>
Reviewed-by: Keqian Zhu <zhukeqian1@huawei.com>
Acked-by: Xie Xiuqi <xiexiuqi@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/arm/kernel/paravirt.c   | 2 +-
 arch/arm64/kernel/paravirt.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/kernel/paravirt.c b/arch/arm/kernel/paravirt.c
index 4cfed91fe256..3c34f456e400 100644
--- a/arch/arm/kernel/paravirt.c
+++ b/arch/arm/kernel/paravirt.c
@@ -15,4 +15,4 @@ struct static_key paravirt_steal_enabled;
 struct static_key paravirt_steal_rq_enabled;
 
 struct paravirt_patch_template pv_ops;
-EXPORT_SYMBOL_GPL(pv_ops);
+EXPORT_SYMBOL(pv_ops);
diff --git a/arch/arm64/kernel/paravirt.c b/arch/arm64/kernel/paravirt.c
index 847b3c8b1218..ecbd2bf2e6fb 100644
--- a/arch/arm64/kernel/paravirt.c
+++ b/arch/arm64/kernel/paravirt.c
@@ -38,7 +38,7 @@ struct paravirt_patch_template pv_ops = {
 #endif
 	.lock.vcpu_is_preempted		= __native_vcpu_is_preempted,
 };
-EXPORT_SYMBOL_GPL(pv_ops);
+EXPORT_SYMBOL(pv_ops);
 
 struct pv_time_stolen_time_region {
 	struct pvclock_vcpu_stolen_time *kaddr;
-- 
2.26.2

